osquery/tables/yara/yara.cpp

```diff
@@ -177,6 +222,21 @@ Status getYaraRules(YARAConfigParser parser,
       break;
     }
 
+    case YC_URL: {
```

Comment:
Speaking of caching... This is all inside the compiled cache block, isn't it? Which, is probably desired to keep from re-compiling these rules. /However/ doesn't that mean that the URL content will be cached for the life of the osquery process?

Reply:
The compiled cache blocks gets cleaned for `YC_RULE` and `YC_URL` type after the scan and will not be there for the life-time of the osquery process.

```  // Rule string is hashed before adding to the cache. There are
  // possibilities of collision when arbitrary queries are executed
  // with distributed API. Clear the hash string from the cache
  // Also cleanup the cache block if rules are downloaded from url
  for (const auto& sign : scanContext) {
    if (sign.first == YC_RULE || sign.first == YC_URL) {
      auto it = rules.find(hashStr(sign.second, sign.first));
      if (it != rules.end()) {
        rules.erase(hashStr(sign.second, sign.first));
      }
    }
  }
```

Reply:
Ah. Hrm. I know that's in a different PR, but it seems weird.

If we're storing these with a key based on the hash of the rule body, why delete? (I know it's existing code)

Reply:
The reason I throw them out of the cache after scanning is to avoid having the recompiled rule for the lifetime of the osquery process. It fits with the URL, if we are going to re-fetch the updated contents.